server/src/protocol-translation.ts

```diff
@@ -18,10 +18,27 @@ export function uriToPath(stringUri: string): string | undefined {
     return uri.fsPath;
 }
 
+// Identify yarn PnP archive files (.zip)
+const archivePathRE = /^\/zipfile:/;
```

Comment:
This all feels a bit gross and not sure how it'll play with windows paths but here it is anyway 😬

Reply:
I think we can drop these changes if we get a change merged into yarn that just exposes the resolved virtual module's on disk path, like this

https://github.com/tapayne88/berry/commit/ed189a9cbd97fe94e066b0a10e3b8c56d3c2b97d